File: Lab_6/main.py
# main.py
from biblioteca import Biblioteca
from libro import Libro

biblioteca = Biblioteca()
    
while True:
    print("\n--- Menu de Biblioteca ---")
    print("1. Agregar libro")
    print("2. Mostrar libros")
    print("3. Prestar libro")
    print("4. Devolver libro")
    print("5. Salir")
    opcion = input("Elige una opcion: ")

    if opcion == "1":
        titulo = input("Introduce el titulo del libro: ")
        autor = input("Introduce el autor del libro: ")
        isbn = input("Introduce el ISBN del libro: ")
        biblioteca.agregar_libro(Libro(titulo, autor, isbn))
        print("Libro agregado con exito.")

    elif opcion == "2":
        print("\n--- Lista de libros en la biblioteca ---")
        biblioteca.mostrar_libros()

    elif opcion == "3":
        isbn = input("Introduce el ISBN del libro a prestar: ")
        if biblioteca.prestar_libro(isbn):
            print("Libro prestado con exito.")
        else:
            print("El libro no esta disponible para prestamo o no existe.")

    elif opcion == "4":
        isbn = input("Introduce el ISBN del libro a devolver: ")
        if biblioteca.devolver_libro(isbn):
            print("Libro devuelto con exito.")
        else:
            print("El libro no estaba prestado o no existe.")

    elif opcion == "5":
        print("Saliendo del programa...")
        break

    else:
        print("Opcion no valida, intenta de nuevo.")

File: Lab_6/Main.java
// Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Biblioteca biblioteca = new Biblioteca();
        Scanner scanner = new Scanner(System.in);
        int opcion;

        do {
            System.out.println("\n--- Menú de Biblioteca ---");
            System.out.println("1. Agregar libro");
            System.out.println("2. Mostrar libros");
            System.out.println("3. Prestar libro");
            System.out.println("4. Devolver libro");
            System.out.println("5. Salir");
            System.out.print("Elige una opción: ");
            opcion = scanner.nextInt();
            scanner.nextLine(); // Consumir la nueva línea

            switch (opcion) {
                case 1:
                    System.out.print("Introduce el título del libro: ");
                    String titulo = scanner.nextLine();
                    System.out.print("Introduce el autor del libro: ");
                    String autor = scanner.nextLine();
                    System.out.print("Introduce el ISBN del libro: ");
                    String isbn = scanner.nextLine();
                    biblioteca.agregarLibro(new Libro(titulo, autor, isbn));
                    System.out.println("Libro agregado con éxito.");
                    break;

                case 2:
                    System.out.println("\n--- Lista de libros en la biblioteca ---");
                    biblioteca.mostrarLibros();
                    break;

                case 3:
                    System.out.print("Introduce el ISBN del libro a prestar: ");
                    String isbnPrestar = scanner.nextLine();
                    if (biblioteca.prestarLibro(isbnPrestar)) {
                        System.out.println("Libro prestado con éxito.");
                    } else {
                        System.out.println("El libro no está disponible para préstamo o no existe.");
                    }
                    break;

                case 4:
                    System.out.print("Introduce el ISBN del libro a devolver: ");
                    String isbnDevolver = scanner.nextLine();
                    if (biblioteca.devolverLibro(isbnDevolver)) {
                        System.out.println("Libro devuelto con éxito.");
                    } else {
                        System.out.println("El libro no estaba prestado o no existe.");
                    }
                    break;

                case 5:
                    System.out.println("Saliendo del programa...");
                    break;

                default:
                    System.out.println("Opción no válida, intenta de nuevo.");
                    break;
            }
        } while (opcion != 5);

        scanner.close();
    }
}
